sklearn/metrics/cluster/supervised.py

```diff
@@ -601,7 +601,7 @@ def mutual_info_score(labels_true, labels_pred, contingency=None):
     log_contingency_nm = np.log(nz_val)
     contingency_nm = nz_val / contingency_sum
     # Don't need to calculate the full outer product, just for non-zeroes
-    outer = pi.take(nzx) * pj.take(nzy)
+    outer = pi.take(nzx).astype(np.int64) * pj.take(nzy).astype(np.int64)
```

Comment:
I would cast before

Reply:
```python 
pi = np.ravel(contingency.sum(axis=1, dtype=np.int64))
pj = np.ravel(contingency.sum(axis=0, dtype=np.int64))
```

Reply:
This does not work for all NumPy versions. And not worth to backport the feature for one line. Hence switched back to casting it later.